contracts/vaults/BalancerComposableAuraVault.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.17;
+
+import {NotionalProxy} from "../../interfaces/notional/NotionalProxy.sol";
+import {Deployments} from "../global/Deployments.sol";
+import {BalancerSpotPrice} from "./balancer/BalancerSpotPrice.sol";
+import {
+    AuraStakingMixin,
+    AuraVaultDeploymentParams,
+    DeploymentParams
+} from "./balancer/mixins/AuraStakingMixin.sol";
+import {IComposablePool} from "../../interfaces/balancer/IBalancerPool.sol";
+import {IBalancerVault} from "../../interfaces/balancer/IBalancerVault.sol";
+
+contract BalancerComposableAuraVault is AuraStakingMixin {
+    /// @notice Helper singleton function to calculate spot prices
+    BalancerSpotPrice immutable SPOT_PRICE;
+
+    constructor(
+        NotionalProxy notional_,
+        AuraVaultDeploymentParams memory params,
+        BalancerSpotPrice _spotPrice
+    ) AuraStakingMixin(notional_, params) {
+        // BPT_INDEX must be defined for a composable pool
+        require(BPT_INDEX != NOT_FOUND);
+        SPOT_PRICE = _spotPrice;
+    }
+
+    /// @notice strategy identifier
+    function strategy() external override pure returns (bytes4) {
+        return bytes4(keccak256("BalancerComposableAuraVault"));
+    }
+
+    function _joinPoolAndStake(
+        uint256[] memory amounts, uint256 minPoolClaim
+    ) internal override returns (uint256 lpTokens) {
+        // Composable pool custom data does not include the BPT token amount so 
+        // we loop here to remove it from the customData
+        uint256[] memory amountsWithoutBpt = new uint256[](amounts.length - 1);
+        uint256 j;
+        for (uint256 i; i < amounts.length; i++) {
+            if (i == BPT_INDEX) continue;
+            amountsWithoutBpt[j++] = amounts[i];
```

Comment:
Does this for loop work correctly? It looks like the first index that will be written on amountsWithoutBpt will be 1, and the 0 index will be left blank unless I misunderstand the syntax.

Reply:
The syntax is correct but a little too cute:
![image](https://github.com/notional-finance/leveraged-vaults/assets/977434/c265c1b4-0922-47db-a107-54610a01df29)